src/ConfirmationTooltip/ConfirmationTooltip.story.tsx

```diff
@@ -1,21 +1,33 @@
 import React from "react";
+import userEvent from "@testing-library/user-event";
 import { Button } from "../Button";
 import { ConfirmationTooltip } from "../ConfirmationTooltip";
+import { findByText } from "@testing-library/dom";
+import { PerformUserInteraction } from "../shared/PerformUserInteraction";
 import { storiesOf } from "@storybook/react";
 
 storiesOf("ConfirmationTooltip", module)
-  .add("forced visible", () => {
-    return (
-      <div style={{ padding: 50 }}>
-        <ConfirmationTooltip
-          content="default content"
-          forceVisibleForTestingOnly
+  .add(
+    "forced visible",
+    () => {
+      return (
+        <PerformUserInteraction
```

Comment:
Interesting, will these always trigger the new state before the snapshot is taken? I like this a lot better than testing only props. Would love to use this pattern in agm as well

Reply:
This will be incorporated into AGM in https://github.com/mdg-private/engine-frontend/pull/2016.

While I think this is a step in the right direction, it is not an ideal solution still. Chromatic does not have an async rendering feature (https://docs.chromaticqa.com/resource-loading#asynchronous-rendering), so we have to manually add `{ chromatic: { delay: number } }` as the second argument to `.add`. @timbotnik suggested I write a blog post on this and gently nudge the chromatic team to offer the ability to programmatically delay snapshots until something has happened, which I think we be awesome.

All that being said, I'm ok adding a 2 second delay to a chromatic capture since they all run in parallel and this would enable us to write less magic props to force component states.

Reply:
To answer your question more succinctly 

> will these always trigger the new state before the snapshot is taken

No, it will not. We have to manually delay the snapshot with a best-guess time.

Reply:
Ah okay, that makes more sense. I still really like this setup though! Wasnt a huge fan of test props for stories